test/unit/UpdatesPublisherTest.js

```diff
@@ -0,0 +1,114 @@
+  createProducerStub.onFirstCall().returns(Promise.reject(new Error('expected rejection')))
+  createProducerStub.onSecondCall().returns(Promise.resolve())
+  await publisher.setupProducer()
+  t.true(createProducerStub.calledTwice)
+})
+test.serial('#setupChanges', async (t) => {
+  const {producerOptions, sandbox, publisher} = t.context
+  publisher.configure(producerOptions, 'http://unhappypath.sad', 'http://unhappierpath.sad', console)
+  const consoleSpy = sandbox.spy(console, 'info')
+  await publisher.setupChanges()
+  t.true(consoleSpy.calledWith('changes stream successfully created'))
+  publisher.changes.destroy()
+})
+test.serial('#start', async (t) => {
+  // setup all mocks and context
+  const {producerOptions, sandbox, publisher} = t.context
+  publisher.configure(producerOptions, 'http://unhappypath.sad', 'http://unhappierpath.sad', console)
+  publisher.producer = new MockProducer()
+  publisher.changes = new MockChangesStream()
+  // stubbing creation methods
+  const setupProducerStub = sandbox.stub(publisher, 'setupProducer')
+  const setupChangesStub = sandbox.stub(publisher, 'setupChanges')
+  setupProducerStub.resolves()
+  setupChangesStub.resolves()
+  // spying mocks' methods
+  const producerStopSpy = sandbox.spy(publisher.producer, 'stop')
+  const changesDestroySpy = sandbox.spy(publisher.changes, 'destroy')
+  // launch the publisher event loop
+  publisher.start()
+  t.true(setupProducerStub.notCalled)
+  t.true(setupChangesStub.notCalled)
+  // first producer fails
+  publisher.producer.emitError()
+  await Promise.delay(1100)
+  t.true(producerStopSpy.calledOnce)
+  t.true(changesDestroySpy.calledOnce)
+  t.true(setupChangesStub.calledOnce)
+  t.true(setupProducerStub.calledOnce)
+  // then changes fails
+  publisher.changes.emitError()
+  await Promise.delay(1100)
+  t.true(producerStopSpy.calledOnce) // i.e not called
+  t.true(setupProducerStub.calledOnce) // ^
+  console.log(changesDestroySpy.callCount)
+  t.true(changesDestroySpy.callCount > 1)
+  t.true(setupChangesStub.callCount > 1)
+  // finally changes detects changes
+  const producerSendMessageSpy = sandbox.spy(publisher.producer, 'sendMessage')
+  publisher.changes.emitData()
+  await Promise.delay(100)
+  t.true(producerSendMessageSpy.calledWith({}))
+})
+
+test.serial('#stop', async (t) => {
+  const {producerOptions, sandbox, publisher} = t.context
+  publisher.configure(producerOptions, 'http://unhappypath.sad', 'http://unhappierpath.sad', console)
+  publisher.producer = new MockProducer()
+  publisher.changes = new MockChangesStream()
+  const producerStopSpy = sandbox.spy(publisher.producer, 'stop')
+  const changesDestroySpy = sandbox.spy(publisher.changes, 'destroy')
+  publisher.start()
+  t.true(producerStopSpy.notCalled)
+  t.true(changesDestroySpy.notCalled)
+  await publisher.stop()
+  t.true(producerStopSpy.calledOnce)
+  t.true(changesDestroySpy.calledOnce)
+})
```

Comment:
Some spacing in between major parts of these tests would help a lot with readability/maintainability. You had good comments in the test above this one, but the others are lacking some comments that would make them easier to follow.